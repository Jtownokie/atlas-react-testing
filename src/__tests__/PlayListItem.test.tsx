import { render } from "@testing-library/react";
import { expect, test, describe } from "vitest";
import PlayListItem from "../components/PlayListItem";

describe("Testing for PlayListItem Component", () => {
  test("PlayListItem renders correctly", () => {
    const { container } = render(
      <PlayListItem
        title="Song Title"
        artist="Artist"
        backgroundColor="bg-light-burgundy"
        songLength="5:45"
        onClick={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("PlayListItem renders correctly with empty strings", () => {
    const { container } = render(
      <PlayListItem
        title=""
        artist=""
        backgroundColor=""
        songLength=""
        onClick={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("PlayListItem correctly applies background", () => {
    const { container, getByText } = render(
      <PlayListItem
        title="Song Title 1"
        artist="Artist 1"
        backgroundColor="bg-blue"
        songLength="5:45 1"
        onClick={() => {}}
      />,
    );
    const titleElement = getByText("Song Title 1");
    const backgroundDiv = titleElement.closest("div")?.parentElement;
    expect(backgroundDiv).toHaveClass("bg-blue");
    expect(container).toMatchSnapshot();
  });
});
