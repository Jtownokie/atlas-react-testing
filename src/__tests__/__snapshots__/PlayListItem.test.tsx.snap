// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem correctly applies background 1`] = `
<div>
  <div
    class="mb-1 flex cursor-pointer flex-row items-center justify-between rounded-md bg-blue"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-sm font-medium text-gold"
      >
        Song Title 1
      </p>
      <p
        class="text-sm font-medium text-cyan"
      >
        Artist 1
      </p>
    </div>
    <p
      class="text-sm font-medium text-cyan"
    >
      5:45 1
    </p>
  </div>
</div>
`;

exports[`PlayListItem renders correctly 1`] = `
<div>
  <div
    class="mb-1 flex cursor-pointer flex-row items-center justify-between rounded-md bg-light-burgundy"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-sm font-medium text-gold"
      >
        Song Title
      </p>
      <p
        class="text-sm font-medium text-cyan"
      >
        Artist
      </p>
    </div>
    <p
      class="text-sm font-medium text-cyan"
    >
      5:45
    </p>
  </div>
</div>
`;

exports[`PlayListItem renders correctly with empty strings 1`] = `
<div>
  <div
    class="mb-1 flex cursor-pointer flex-row items-center justify-between rounded-md "
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-sm font-medium text-gold"
      />
      <p
        class="text-sm font-medium text-cyan"
      />
    </div>
    <p
      class="text-sm font-medium text-cyan"
    />
  </div>
</div>
`;
